Admin/Test-AMSI.ps1

```diff
@@ -118,23 +212,52 @@ function Test-AMSI {
         }
         return
     }
+
     if ($CheckAMSIProviders) {
+        CheckWindowsVersionIsW16orOlder
         $AMSI = Get-ChildItem 'HKLM:\SOFTWARE\Microsoft\AMSI\Providers' -Recurse
         $AMSI -match '[0-9A-Fa-f\-]{36}' | Out-Null
         $Matches.Values | ForEach-Object { Get-ChildItem "HKLM:\SOFTWARE\Classes\ClSid\{$_}" | Format-Table -AutoSize }
     }
+
+    $getSO = $null
     if ($EnableAMSI) {
-        Remove-SettingOverride -Identity DisablingAMSIScan -Confirm:$false
-        Get-ExchangeDiagnosticInfo -Process Microsoft.Exchange.Directory.TopologyService -Component VariantConfiguration -Argument Refresh
-        Write-Warning "Remember to restart IIS for this to take affect. You can accomplish this by running .\Test-AMSI.ps1 -RestartIIS"
+        CheckExchangeAndWindowsVersionsSupportedByAMSI
+        $getSO = Get-SettingOverride -Identity DisablingAMSIScan -ErrorAction SilentlyContinue | Where-Object { $_.SectionName -eq 'HttpRequestFiltering' -and $_.Parameters -eq 'Enabled=False' -and $_.Reason -eq 'Disabled via CSS-Exchange Script' }
```

Comment:
I wouldn't include the Reason in this Where clause. It doesn't really matter if the Reason says it was disabled by the script, or if the user did it manually, or whatever. If we have this setting there, it's already set, regardless of Reason.